### PR TITLE
doc: Added msgr on-wire-compression

### PR DESCRIPTION
**Pull request by giladmaya**
(doc) Added msgr on-wire-compression (feature introduces in #36517)

### CODE REVIEW

**Comment by JoshSalomon** [comment]
@rzarzynski - any news?

File: doc/dev/msgr2.rst [hunk 1]
```diff
@@ -528,6 +533,62 @@ where epilogue is::
 
 late_status has the same meaning as in msgr2.1-crc mode.
 
+Compression
+-----------
+Compression handshake is implemened using msgr2 feature-based handshaking.
```
Comment (rzarzynski) [inline doc/dev/msgr2.rst:538]:
```suggestion
Compression handshake is implemented using msgr2 feature-based handshaking.
```

File: doc/dev/msgr2.rst [hunk 2]
```diff
@@ -528,6 +533,62 @@ where epilogue is::
...
+In this phase, the peers exchange their supported compression methods and if compression is desired.
```
Comment (rzarzynski) [inline doc/dev/msgr2.rst:539]: How about rephrasing this line? _the peers exchange (...) and if compression is desired_ is hard to grasp.

File: doc/rados/configuration/msgr2.rst [hunk 3]
```diff
@@ -95,10 +95,10 @@ Similarly, two options control whether IPv4 and IPv6 addresses are used:
    not yet tested as of Nautilus v14.2.0 and likely needs some
    additional code changes to work correctly.
 
-Connection modes
-----------------
+Authentication modes
+--------------------
 
-The v2 protocol supports two connection modes:
+The v2 protocol supports two authentication modes:
```
Comment (rzarzynski) [inline doc/rados/configuration/msgr2.rst:101]: I doubt this replacement is an improvement. These modes are far more than just authentication (they include e.g. data intgrity0. I propose to leave this intact or dissect into a separated PR.

File: doc/rados/configuration/msgr2.rst [hunk 4]
```diff
@@ -146,6 +146,41 @@ secure) requirements on communication with the monitors.
...
+
+
+Compression mode configuration options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For all connections, there is an option that controls compression usage in secure mode
+
+.. confval:: ms_compress_secure
+
+There are a parallel set of options that apply specifically to osds,
```
Comment (rzarzynski) [inline doc/rados/configuration/msgr2.rst:177]:
```suggestion
There is a parallel set of options that apply specifically to OSDs,
```

File: doc/rados/configuration/msgr2.rst [hunk 5]
```diff
@@ -146,6 +146,41 @@ secure) requirements on communication with the monitors.
...
+
+Compression mode configuration options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For all connections, there is an option that controls compression usage in secure mode
+
+.. confval:: ms_compress_secure
+
+There are a parallel set of options that apply specifically to osds, 
+allowing administrators to set different requirements on communication between osds.
```
Comment (rzarzynski) [inline doc/rados/configuration/msgr2.rst:178]:
```suggestion
allowing administrators to set different requirements on communication between OSDs.
```

File: doc/rados/configuration/msgr2.rst [hunk 6]
```diff
@@ -146,6 +146,41 @@ secure) requirements on communication with the monitors.
...
 
+
+Compression modes
+-----------------
+
+The v2 protocol supports two compression modes:
+
+* *force* mode provides:
+
+  - In multi availability zones deployment, compressing replication messages between osds saves latency.
```
Comment (rzarzynski) [inline doc/rados/configuration/msgr2.rst:157]:
```suggestion
  - In multi-availability zones deployment, compressing replication messages between OSDs saves latency.
```

**Review by rzarzynski — CHANGES_REQUESTED** [review]

**Comment by rzarzynski** [comment]
Generally LGTM. Just one thing + a bunch of minors / nits.

**Review by rzarzynski — APPROVED** [review]
@JoshSalomon: LGTM. There is a minor typo. It would be preferable to have it fixed before the merge but I would be also fine with a follow-up fix.

**Pull request merged**